app/helpers/page-guard.js

```diff
@@ -13,11 +14,93 @@ function guardPage (request, guardData, rule = null) {
   const isServiceDecommissioned = (request.url.pathname !== startPageUrl && currentUrl !== 'login' && serviceDecommissioned)
   if (isServiceDecommissioned) return isServiceDecommissioned
   if (guardData) {
-    if (rule) {
-      result = rule.condition === 'ANY' && !guardData.some(dependcyKey => getYarValue(request, dependcyKey) !== null)
-    } else {
-      result = guardData.filter(dependcyKey => getYarValue(request, dependcyKey) === null).length > 0
+    // filter list of answers with keys?
+
+    // make an object of question as key and answers for said question?
+
+    let preValidationList = []
+
+    for(let i=0; i<guardData.preValidationKeys.length; i++) {
+      preValidationList.push({
+       key: guardData.preValidationKeys[i], 
+       values: (guardData.preValidationAnswer.filter((answer) => answer.startsWith(guardData.preValidationUrls[i]))),
+       url: guardData.preValidationUrls[i]
+      }
+      );
     }
+
+    console.log('should be a formatted list combining all answewrs and relevant values in one object', preValidationList)
+
+    // should format preValidations as below
```

Comment:
do we need a console log here

Reply:
nope, its for my tesating when im able to run the app, to check what the key/value object looks like when its generated